ksql-functional-tests/src/test/java/io/confluent/ksql/test/TopologyFileGenerator.java

```diff
@@ -145,58 +140,76 @@ private static void writeExpectedTopologyFiles(
-                    buildQuery(testCase, serviceContext, ksqlEngine, ksqlConfig);
-                final Map<String, String> configsToPersist
-                    = new HashMap<>(ksqlConfig.getAllConfigPropsWithSecretsObfuscated());
-
-                // Ignore the KStreams state directory as its different every time:
-                configsToPersist.remove("ksql.streams.state.dir");
-
-                ExpectedTopologiesTestLoader.writeExpectedTopologyFile(
-                    testCase.getName(),
-                    queryMetadata,
-                    configsToPersist,
-                    objectWriter,
-                    topologyDir);
-            }
-        });
+        testCases.forEach(testCase -> writeExpectedToplogyFile(topologyDir, testCase));
     }
 
-    static Map<String, Object> baseConfig() {
-        return ImmutableMap.<String, Object>builder()
-            .put(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, "localhost:0")
-            .put(ConsumerConfig.AUTO_COMMIT_INTERVAL_MS_CONFIG, 0)
-            .put(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "earliest")
-            .put(StreamsConfig.CACHE_MAX_BYTES_BUFFERING_CONFIG, 0)
-            .put(StreamsConfig.STATE_DIR_CONFIG, TestUtils.tempDirectory().getPath())
-            .put(KsqlConfig.KSQL_SERVICE_ID_CONFIG, "some.ksql.service.id")
-            .put(KsqlConfig.KSQL_USE_NAMED_INTERNAL_TOPICS,
-                KsqlConfig.KSQL_USE_NAMED_INTERNAL_TOPICS_ON)
-            .put(StreamsConfig.TOPOLOGY_OPTIMIZATION, "all")
-            .build();
+    private static void writeExpectedToplogyFile(final Path topologyDir, final TestCase testCase) {
+        try {
+            final Path topologyFile = buildExpectedTopologyPath(topologyDir, testCase);
+
+            final String topologyContent = buildExpectedTopologyContent(testCase, Optional.empty());
+
+            Files.write(topologyFile,
+                topologyContent.getBytes(StandardCharsets.UTF_8),
+                StandardOpenOption.CREATE,
+                StandardOpenOption.WRITE,
+                StandardOpenOption.TRUNCATE_EXISTING
+            );
+        } catch (final IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    static Path buildExpectedTopologyPath(final Path topologyDir, final TestCase testCase) {
+        return ExpectedTopologiesTestLoader.buildExpectedTopologyPath(
+            testCase.getName(),
+            topologyDir
+        );
+    }
+
+    static String buildExpectedTopologyContent(
+        final TestCase testCase,
+        final Optional<Map<String, ?>> persistedConfigs
+    ) {
+        final KsqlConfig baseConfigs = new KsqlConfig(TestExecutor.baseConfig())
+            .cloneWithPropertyOverwrite(testCase.properties());
+
+        final KsqlConfig ksqlConfig = persistedConfigs
+            .map(baseConfigs::overrideBreakingConfigsWithOriginalValues)
+            .orElse(baseConfigs);
+
+        try (final ServiceContext serviceContext = getServiceContext();
```

Comment:
nit: final on new line? to be consistent with our weird and ugly function parameter on new lines coding standard? ;)

Reply:
Nope, this isn't a parameter...